File: vivid/src/main/java/com/icetea09/vivid/camera/CameraModule.java
package com.icetea09.vivid.camera;

import android.content.Context;
import android.content.Intent;

import com.icetea09.vivid.imagepicker.Configuration;

public interface CameraModule {
    Intent getCameraIntent(Context context, Configuration config);

    void getImage(Context context, Intent intent, OnImageReadyListener imageReadyListener);
}

File: library/src/main/java/com/icetea09/vivid/camera/CameraModule.kt
package com.icetea09.vivid.camera

import android.content.Context
import android.content.Intent

import com.icetea09.vivid.imagepicker.Configuration

interface CameraModule {
    fun getCameraIntent(context: Context, config: Configuration): Intent?

    fun getImage(context: Context, intent: Intent, imageReadyListener: OnImageReadyListener?)
}
